clang-tools-extra/clangd/ClangdServer.cpp

```diff
@@ -624,6 +624,50 @@ void ClangdServer::rename(PathRef File, Position Pos, llvm::StringRef NewName,
   WorkScheduler->runWithAST("Rename", File, std::move(Action));
 }
 
+void ClangdServer::indexedRename(
+    std::map<URIForFile, std::vector<Position>> Positions, PathRef PrimaryFile,
+    llvm::StringRef OldName, llvm::StringRef NewName, Callback<FileEdits> CB) {
+  ParseInputs Inputs;
+  Inputs.TFS = &TFS;
+  Inputs.CompileCommand = CDB.getCompileCommand(PrimaryFile)
+                              .value_or(CDB.getFallbackCommand(PrimaryFile));
+  IgnoreDiagnostics IgnoreDiags;
+  std::unique_ptr<CompilerInvocation> CI =
+      buildCompilerInvocation(Inputs, IgnoreDiags);
+  if (!CI) {
+    return CB(llvm::make_error<llvm::StringError>(
+        "Unable to get compiler arguments for primary file",
+        llvm::inconvertibleErrorCode()));
+  }
+  const LangOptions &LangOpts = CI->getLangOpts();
+
+  tooling::SymbolName OldSymbolName(OldName, LangOpts);
+  tooling::SymbolName NewSymbolName(NewName, LangOpts);
+
+  llvm::StringMap<std::vector<Range>> FilesToRanges;
+  for (auto Entry : Positions) {
+    std::vector<Range> Ranges;
```

Comment:
```suggestion
    std::vector<Range> &Ranges = FilesToRanges[Entry.first.file()];
```
And then no need for the `FilesToRanges[Entry.first.file()] = Ranges;` below. Just avoids an extra copy.